deprecation-notice.php

```diff
@@ -0,0 +1,47 @@
+<?php
+// Add an admin notice for outdated plugin version
+add_action('admin_notices', 'notify_plugin_update');
+function notify_plugin_update() {
+    $dismissed_timestamp = get_option('onesignal_deprecation_notice_dismissed_time');
+    $time_passed = !$dismissed_timestamp || (time() - $dismissed_timestamp) > (7 * DAY_IN_SECONDS);
+    $cutoff_date = strtotime('2025-12-31 23:59:59'); // Set cutoff date and time
+
+    // Show the notice only if it's before the cutoff date and the notice has not been recently dismissed
+    if (time() <= $cutoff_date && $time_passed) {
+        echo '
+        <div class="notice notice-warning is-dismissible" id="onesignal-update-notice">
+            <p>
+              <strong>
+                OneSignal Push Important Update:</strong>
+                All push notification configuration options are moving to OneSignal.com.
+                Please ensure your settings are updated by December 13, 2024 to avoid any service disruptions.
+                <a href="https://documentation.onesignal.com/docs/wordpress-plugin-30">Learn More.</a>
```

Comment:
Add target="_blank" to anchor, so not to navigate away from their site.

Reply:
Good call